### ISSUE TITLE
Support adding license header in .txt, .patch and .tmpl files

### ISSUE THREAD

**Issue by maqiuyujoyce** [google/addlicense#106]
Hi, can `addlicense` support adding the license header to .txt, .patch and .tmpl files? Or is it intentional to not support these use cases?

We want to use `addlicense` to add the license header to various types of files automatically and we're willing to contribute if it is not a design decision to not support them.

**Comment by bryanloz-xilinx** [comment]
Agree, this would be very useful for common files such as CMakeLists.txt where `#` can be used for comment.

**Comment by willnorris** [comment]
addlicense already handles cmakelists.txt files, or at least it's supposed to: https://github.com/google/addlicense/blob/master/main.go#L311-L314 Does that not work?

We certainly wouldn't want to add headers to *all* .txt files, which is why cmakelists.txt is handled the way it is (looking for the full name)

I'm not sure what a .tmpl file would even be... I worry that that is used to generically mean "template file" for all sorts of underlying file types.  Is it an HTML file?  Something else?

.patch could possibly work, but aren't there different patch formats?  Do they all support the same comment style?

**Comment by bryanloz-xilinx** [comment]
@willnorris it wasn't working for me. Maybe there is a case sensitivity issue here? I don't know why, but historically CMake projects use `CMakeLists.txt` by convention.

P.S. thanks for providing this tool. It saved me a lot of work even though it didn't work exactly how I wanted it to.

**Comment by basickarl** [comment]
I just spent an hour trying to figure out why nothing was being added. This needs to be more clear in the documentation guys.
